lumicks/pylake/calibration.py

```diff
@@ -1,3 +1,218 @@
+            calib_slice = f.force1x[calibration]
+
+            # De-calibrate to volts using the calibration that was active before this slice.
+            previous_calibration = calib_slice.calibration[0]
+            calib_slice = calib_slice / previous_calibration.force_sensitivity
+
+            power_spectrum_params = previous_calibration.power_spectrum_params()
+            power_spectrum = lk.calculate_power_spectrum(calib_slice.data, **power_spectrum_params)
+            power_spectrum.plot()
+        """
+        self._verify_full()
+
+        return {
+            "num_points_per_block": self.num_points_per_block,
+            "sample_rate": self.sample_rate,
+            "excluded_ranges": self.excluded_ranges,
+            "fit_range": self.fit_range,
+        }
+
+    def _model_params(self):
+        """Returns parameters with which to create an active or passive calibration model"""
+
+        # TODO: Model needs to support fixed_diode and fixed_alpha, keep API private until finalized
+        self._verify_full()
+        model_params = _read_from_bl_dict(self.data)
+        model_params["sample_rate"] = self.sample_rate
+        model_params["fast_sensor"] = self.fast_sensor
+        model_params["axial"] = bool(self.data.get("Axial calibration"))
+        model_params["hydrodynamically_correct"] = bool(
+            self.data.get("Hydrodynamic correction enabled")
+        )
+        return model_params
+
+    def calibration_params(self):
+        """Returns parameters to calculate the same calibration as the one performed by Bluelake
+
+        Examples
+        --------
+        ::
+
+            import lumicks.pylake as lk
+
+            f = lk.File("passive_calibration.h5")
+            calibration = f.force1x.calibration[1]  # Grab a calibration item for force 1x
+
+            # Slice the data corresponding to the calibration we want to reproduce.
+            calib_slice = f.force1x[calibration]
+
+            # De-calibrate to volts using the calibration that was active before this slice.
+            previous_calibration = calib_slice.calibration[0]
+            calib_slice = calib_slice / previous_calibration.force_sensitivity
+
+            calibration_params = previous_calibration.calibration_params()
+            new_calibration = lk.calibrate_force(calib_slice.data, **calibration_params)
+            new_calibration.plot()
+
+            # Make a new calibration, but change the amount of blocking
+            less_blocking_params = calibration_params | {"num_points_per_block": 200}
+            less_blocking = lk.calibrate_force(calib_slice.data, **less_blocking_params)
+            less_blocking.plot()
+        """
+        model_params = self._model_params()
+        model_params["active_calibration"] = self.active_calibration
+
+        def check_defined(pair):
+            return pair[1] is not None
+
+        return dict(filter(check_defined, (self.power_spectrum_params() | model_params).items()))
```

Comment:
interestingly, I think the `black` formatting for a longer line using a `lambda` might actually make this more readable

```suggestion
        return dict(
            filter(
                lambda pair: pair[1] is not None,
                (self.power_spectrum_params() | model_params).items(),
            )
        )
```

Reply:
Yeah, I agree. But considering the change to `_read_from_bl_dict`, filtering at the end is no longer needed.